src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -2663,11 +2670,23 @@ public RegistryAuth registryAuth() {
      * @return Builder
      */
     public Builder registryAuth(final RegistryAuth registryAuth) {
-      this.registryAuth = registryAuth;
+      this.registryAuthCalled = true;
+      this.registryAuthSupplier = new NoOpRegistryAuthSupplier(registryAuth);
+      return this;
+    }
+
+    public Builder registryAuthSupplier(final RegistryAuthSupplier registryAuthSupplier) {
+      this.registryAuthCSupplierCalled = true;
+      this.registryAuthSupplier = registryAuthSupplier;
       return this;
     }
 
     public DefaultDockerClient build() {
+      if (registryAuthCalled && registryAuthCSupplierCalled) {
+        throw new RuntimeException("LOGIC ERROR: DefaultDockerClient does not support being built "
```

Comment:
also I think you could avoid the need for these two booleans by moving this check into the `registryAuth(RegistryAuth)` and `registryAuthSupplier(RegistryAuthSupplier)` methods - you can just throw an exception of `this.registryAuthSupplier` is already set